index.js

```diff
@@ -169,7 +169,7 @@ function markdownTable(table, options) {
 
     index = -1;
 
-    while (++index < cellCount) {
+    while (settings.straighten !== false && ++index < cellCount) {
```

Comment:
Could you wrap the whole `while` in an `if (settings.straighten !== false)`? No need to compute this on every iteration.